File: sdks/python/src/echo_agent_sdk/model_input_modality_values.py
from __future__ import annotations

from enum import Enum


class ModelInputModality(str, Enum):
    """Content modality accepted by one configured model."""

    TEXT = "text"
    IMAGE = "image"
    AUDIO = "audio"
    VIDEO = "video"

    @classmethod
    def text_only(cls) -> list[ModelInputModality]:
        return [cls.TEXT]

    @classmethod
    def all_supported(cls) -> list[ModelInputModality]:
        return [cls.TEXT, cls.IMAGE, cls.AUDIO, cls.VIDEO]

File: sdks/typescript/src/model_input_modality_values.ts
/** Content modalities accepted by one configured model. */
export const ModelInputModality = Object.freeze({
  Text: "text",
  Image: "image",
  Audio: "audio",
  Video: "video",
  textOnly(): readonly ModelInputModality[] {
    return Object.freeze([this.Text]);
  },
  allSupported(): readonly ModelInputModality[] {
    return Object.freeze([this.Text, this.Image, this.Audio, this.Video]);
  },
} as const);

export type ModelInputModality = "text" | "image" | "audio" | "video";
